contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -791,18 +784,23 @@ func (s *SchedulerServer) bootstrap(hks hyperkube.Interface, sc *schedcfg.Config
 	}
 
 	// create event recorder sending events to the "" namespace of the apiserver
+	eventsClientConfig := *clientConfig
+	eventsClient, err := clientset.NewForConfig(&eventsClientConfig)
+	if err != nil {
+		log.Fatalf("Invalid API configuration: %v", err)
+	}
 	broadcaster := record.NewBroadcaster()
 	recorder := broadcaster.NewRecorder(api.EventSource{Component: api.DefaultSchedulerName})
 	broadcaster.StartLogging(log.Infof)
-	broadcaster.StartRecordingToSink(client.Events(""))
+	broadcaster.StartRecordingToSink(&unversioned_core.EventSinkImpl{eventsClient.Events("")})
```

Comment:
@jdef, sorry for breaking your code. I think it's this line. All the other changes in this PR may be unnecessary. I had encountered the same error when I refactored kubelet and other controllers because of this.